clamd/clamav-daemon.service.in

```diff
@@ -12,6 +12,30 @@ ExecStart=@prefix@/sbin/clamd --foreground=true
 ExecReload=/bin/kill -USR2 $MAINPID
 TimeoutStartSec=420
 
+##
+## Security Hardening Options
+##
+
+# Remove `ProtectSystem`, `ProtectHome`, and `ReadWritePaths`
+# if you want ClamAV to be able to remove infected files.
+ProtectSystem=strict
+ProtectHome=read-only
+ReadWritePaths=/var/log/
+
+NoExecPaths=/
+# If you want to run commands or execute binaries on event,
+# append the full path of the binary or executable to `ExecPaths`
+# Commonly, this is used for `VirusEvent` in clamd.conf or `VirusAction`
+# in clamav-milter.conf.The binaries must be space separated like so:
+;ExecPaths=@prefix@/sbin/clamd /bin/kill /usr/local/bin/send_sms /usr/local/bin/my_infected_message_handler
+ExecPaths=@prefix@/sbin/clamd /bin/kill
```

Comment:
This should work, but I'm seeing:
```
Apr 22 01:52:01 micah-ubuntu-2204 systemd[1]: Started Clam AntiVirus userspace daemon.
Apr 22 01:52:01 micah-ubuntu-2204 systemd[31109]: clamav-daemon.service: Failed to execute /usr/local/sbin/clamd: Permission denied
Apr 22 01:52:01 micah-ubuntu-2204 systemd[31109]: clamav-daemon.service: Failed at step EXEC spawning /usr/local/sbin/clamd: Permission denied
Apr 22 01:52:01 micah-ubuntu-2204 systemd[1]: clamav-daemon.service: Main process exited, code=exited, status=203/EXEC
Apr 22 01:52:01 micah-ubuntu-2204 systemd[1]: clamav-daemon.service: Failed with result 'exit-code'.
```
No idea why.
My generated service file has:
```
[Unit]
Description=Clam AntiVirus userspace daemon
Documentation=man:clamd(8) man:clamd.conf(5) https://docs.clamav.net/
Requires=clamav-daemon.socket
# Check for database existence
ConditionPathExistsGlob=/usr/local/share/clamav/main.{c[vl]d,inc}
ConditionPathExistsGlob=/usr/local/share/clamav/daily.{c[vl]d,inc}

[Service]
ExecStart=/usr/local/sbin/clamd --foreground=true
# Reload the database
ExecReload=/bin/kill -USR2 $MAINPID
TimeoutStartSec=420

##
## Security Hardening Options
##

# Remove `ProtectSystem`, `ProtectHome`, and `ReadWritePaths`
# if you want ClamAV to be able to remove infected files.
ProtectSystem=strict
ProtectHome=read-only
ReadWritePaths=/var/log/

NoExecPaths=/
# If you want to run commands or execute binaries on event,
# append the full path of the binary or executable to `ExecPaths`
# Commonly, this is used for `VirusEvent` in clamd.conf or `VirusAction`
# in clamav-milter.conf.The binaries must be space separated like so:
;ExecPaths=/usr/local/sbin/clamd /bin/kill /usr/local/bin/send_sms /usr/local/bin/my_infected_message_handler
ExecPaths=/usr/local/sbin/clamd /bin/kill

ProtectClock=yes
ProtectKernelTunables=yes
ProtectKernelModules=yes
ProtectKernelLogs=yes
ProtectControlGroups=yes

[Install]
WantedBy=multi-user.target
Also=clamav-daemon.socket
```

Reply:
I have made several changes based on your feedback @micahsnyder and updated this Pull Request. I encountered similar issues to the above but I am now unable to reproduce it. Can you pull my changes and test again, please? I'll keep working on this to see if I find anything else.